### ISSUE TITLE
Taxonomy not preselected in backend form (Underscores in taxonomies break admin input)

### ISSUE THREAD

**Issue by dantleech** [bolt/bolt#5490]
It seems that the form is not prepopulated with the chosen taxonomy value.
1. Add a taxonomy:

``` yaml
region_groups:
    label: Region Group
    slug: region_groups
    singular_slug: region_group
    behaves_like: categories
    options: { regions_nord: "Regions Nord", regions_est: "Regions Est", regions_west: "Regions Ouest", regions_sud: "Regions Sud" }
```
1. Edit the content type with that taxonomy, set "Regions Nord"
2. Save
3. Reload
4. No group is selected, but the record exists in the database:

```
+----+------------+-------------+---------------+--------------+--------------+-----------+
| id | content_id | contenttype | taxonomytype  | slug         | name         | sortorder |
+----+------------+-------------+---------------+--------------+--------------+-----------+
|  5 |         38 | regions     | region_groups | regions_nord | Regions Nord |         0 |
+----+------------+-------------+---------------+--------------+--------------+-----------+
```

**Comment by SahAssar** [comment]
Does this also happen if you don't use underscores in the name/slugs? Just trying to see if this is related/the same issue as #5543

**Comment by SahAssar** [comment]
In my testing it does not happen if one sets the key and the slug to `regiongroups` in your example while keeping everything else the same. So underscores work in the options keys and the singular_slug, but not as the taxonomy key or the slug.

**Comment by SahAssar** [comment]
The great underscore war should be over and fixed in 3.2, so I'm going to close this, but if it turns out this particular battle is still raging feel free to reopen :)
